cli/src/accounts/mod.rs

```diff
@@ -92,44 +122,177 @@ pub async fn create_aze_game_account(player_account_ids: Vec<u64>, small_blind:
+    println!("Consumable notes: {:?}", consumable_notes.len());
+
+    for consumable_note in consumable_notes {
+        let tx_template = TransactionTemplate::ConsumeNotes(account_id, vec![consumable_note.note.id()]);
+        let tx_request = client.build_transaction_request(tx_template).unwrap();
+        execute_tx_and_sync(&mut client, tx_request).await;
+        println!("Waiting...");
+        sleep(Duration::from_secs(5)).await;
+    }
+}
+
+pub async fn enc_action(account_id: AccountId) {
+    // check account storage for a slot
+    // if slot == 1, mask. if slot == 2/3/4, remask
+
+    // 2 mock cards
+    let mut card_points = vec![]; // 2 cards for now
+    card_points.push(keygen(11)); // for now 
+    card_points.push(keygen(12));
+    let pub_key_agg = keygen(123); // for now
+    let masking_factor: u32 = 123;// for now
+
+    if account_id == AccountId::try_from(1085128954612151006).unwrap() { // for now
+        // Mask
+        let cipher_card_1 = mask(pub_key_agg, card_points[0], masking_factor);
+        println!("Cipher card 1 --> {:?}\n", cipher_card_1);
+        let cipher_card_2 = mask(pub_key_agg, card_points[1], masking_factor);
+        println!("Cipher card 2 --> {:?}", cipher_card_2);
+        return
+    }
+
+    // 2 mock masked cards
+    let cipher_card_1 = mask(pub_key_agg, card_points[0], masking_factor);
+    let cipher_card_2 = mask(pub_key_agg, card_points[1], masking_factor);
+
+    // remask
+    let remasked_card_1 = remask(pub_key_agg, cipher_card_1, masking_factor);
+    println!("Remasked card 1 --> {:?}\n", remasked_card_1);
+    let remasked_card_2 = remask(pub_key_agg, cipher_card_2, masking_factor);
+    println!("Remasked card 2 --> {:?}", remasked_card_2);
+
+    // if slot == 2/3, send note to next player
+    // if slot == 4, do nothing as it is the last player
+    send_note(account_id, account_id).await;
+}
+
+pub async fn dec_action(account_id: AccountId) {
+    let players_ids = [317826241474458840, 359196095275670923, 359196095275670923, 359196095275670923];
+    let pub_key_agg = keygen(123); // for now
+    let masking_factor: u32 = 123;// for now
+    let masked_cards = get_mock_cards(); // 2 cards for now
+    // check account storage for a slot
+    // if slot == 1, do final-unmask else inter-unmask
+    if account_id == AccountId::try_from(players_ids[0]).unwrap() {
+        // read masked cards from account storage
+        // 2 mock masked cards for now
+        let final_unmasked_card_1 = final_unmask(pub_key_agg, masked_cards[0].clone(), masking_factor);
+        println!("Final unmasked card 1 --> {:?}\n", final_unmasked_card_1);
+        let final_unmasked_card_2 = final_unmask(pub_key_agg, masked_cards[1].clone(), masking_factor);
+        println!("Final unmasked card 2 --> {:?}", final_unmasked_card_2);
+        return
+    }
+
+    // read masked cards from account storage
+    let unmasked_card_1 = inter_unmask(pub_key_agg, masked_cards[0].clone(), masking_factor);
```

Comment:
same thing, the conditonalization is left right or i am assuming something else ?

Reply:
yes lot of things need to be conditionalized